src/libAtomVM/memory.c

```diff
@@ -253,6 +253,11 @@ unsigned long memory_estimate_usage(term t)
                 t = term_nil();
             }
 
+        } else if (term_is_map(t)) {
```

Comment:
It is not estimating memory usage for key and values, but just for the container. Please take a look the tuple branch.